core/storage/topic/gae_models.py

```diff
@@ -21,6 +21,104 @@
 from google.appengine.ext import ndb
 
 
+class TopicSnapshotMetadataModel(base_models.BaseSnapshotMetadataModel):
+    """Storage model for the metadata for a topic snapshot."""
+    pass
+
+
+class TopicSnapshotContentModel(base_models.BaseSnapshotContentModel):
+    """Storage model for the content of a topic snapshot."""
+    pass
+
+
+class TopicModel(base_models.VersionedModel):
+    """Model for storing Topics.
+
+    This class should only be imported by the topic domain file, the
+    topic services file, and the topic model test file.
+    """
+    SNAPSHOT_METADATA_CLASS = TopicSnapshotMetadataModel
+    SNAPSHOT_CONTENT_CLASS = TopicSnapshotContentModel
+    ALLOW_REVERT = False
+
+    # The name of the topic.
+    name = ndb.StringProperty(required=True, indexed=True)
+    # This includes additional metadata like the description of the topic and
+    # some content paragraphs.
+    description = ndb.StringProperty(indexed=False)
+    # This consists of the list of canonical story ids that are part of
+    # this topic.
+    canonical_story_ids = ndb.StringProperty(repeated=True, indexed=False)
```

Comment:
indexed=True is fine here. We may want to know which topic contains a given story.